hibernate5/src/main/java/com/baeldung/hibernate/proxy/Boss.java

```diff
@@ -0,0 +1,49 @@
+package com.baeldung.hibernate.proxy;
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Entity
+public class Boss implements Serializable {
```

Comment:
Could we change this to "Company"? My reasoning is that it offers a cleaner topological separation between the two entities. In the real work, a boss _is_ an employee, too, so there is a bit more cognitive load when the reader is thinking about a boss and an employee.